src/Nethermind/Nethermind.Network.Stats/Model/InitiateDisconnectReason.cs

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Stats.Model;
+
+/// <summary>
+/// Nethermind level disconnect reason. Don't forget to add the corresponding Eth level disconnect reason in `InitiateDisconnectReasonExtension`.
+/// </summary>
+public enum InitiateDisconnectReason : byte
+{
+    IncomingConnectionRejectedTooManyPeer,
+    SessionAlreadyExist,
+    LostSessionDirectionDecision,
+    OppositeDirectionCleanup,
```

Comment:
Just looking at the names I have no idea what they are about.

Reply:
Maybe `OppositeSessionAlreadyExist` and `ReplacingSessionWithOppositeDirection` ?

Reply:
This `OppositeDirectionCleanup` is very weird. There seems to be multiple place where it cleanup with opposite direction, so I'm thinking it can't actually happen and its just like a cleanup or something. `ReplacingSessionWithOppositeDirection`, not exactly true. That would be `LostSessionDirectionDecision`. Well, I guess that make more sense for it.